src/main/java/org/prebid/server/bidder/tripleliftnative/TripleliftNativeBidder.java

```diff
@@ -82,31 +79,51 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequ
                 errors);
     }
 
-    private Imp modifyImp(Imp imp) throws PreBidException {
+    private Imp modifyImp(Imp imp, BidRequest request) throws PreBidException {
         if (imp.getXNative() == null) {
             throw new PreBidException("no native object specified");
         }
 
-        final ExtImpTriplelift impExt = parseExtImpTriplelift(imp);
-        final String inventoryCode = impExt.getInventoryCode();
-        if (StringUtils.isBlank(inventoryCode)) {
-            throw new PreBidException("no inv_code specified");
-        }
+        final TripleliftNativeExtImp impExt = parseExtImp(imp);
+        final ExtImpTriplelift impExtBidder = impExt.getBidder();
 
         return imp.toBuilder()
-                .tagid(inventoryCode)
-                .bidfloor(impExt.getFloor())
+                .tagid(resolveTagId(request, impExtBidder.getInventoryCode(), impExt.getData()))
+                .bidfloor(impExtBidder.getFloor())
```

Comment:
You need to check whether the provided `impExtBidder.getFloor()` is valid or not. If not, stay with `imp.bidfloor`